tests/store/artifact/test_databricks_artifact_repo.py

```diff
@@ -0,0 +1,137 @@
+# -*- coding: utf-8 -*-
+import os
+
+import pytest
+import mock
+from unittest.mock import ANY
+
+from mlflow.exceptions import MlflowException
+from mlflow.store.artifact.artifact_repository_registry import get_artifact_repository
+from mlflow.store.artifact.dbfs_artifact_repo import DatabricksArtifactRepository
+from mlflow.protos.service_pb2 import ListArtifacts, FileInfo
+from mlflow.protos.databricks_artifacts_pb2 import GetCredentialsForWrite, GetCredentialsForRead, \
+    ArtifactCredentialType, ArtifactCredentialInfo
+
+
+@pytest.fixture()
+def databricks_artifact_repo():
+    return get_artifact_repository('dbfs:/databricks/mlflow-tracking/MOCK-EXP/MOCK-RUN/artifact')
+
+
+DATABRICKS_ARTIFACT_REPOSITORY_PACKAGE = 'mlflow.store.artifact.databricks_artifact_repo'
+DATABRICKS_ARTIFACT_REPOSITORY = DATABRICKS_ARTIFACT_REPOSITORY_PACKAGE + ".DatabricksArtifactRepository"
+
+TEST_FILE_1_CONTENT = u"Hello 🍆🍔".encode("utf-8")
+TEST_FILE_2_CONTENT = u"World 🍆🍔🍆".encode("utf-8")
+TEST_FILE_3_CONTENT = u"¡🍆🍆🍔🍆🍆!".encode("utf-8")
+
+
+@pytest.fixture()
+def test_file(tmpdir):
+    p = tmpdir.join("test.txt")
+    with open(p.strpath, 'wb') as f:
+        f.write(TEST_FILE_1_CONTENT)
+    return p
+
+
+@pytest.fixture()
+def test_dir(tmpdir):
+    with open(tmpdir.mkdir('subdir').join('test.txt').strpath, 'wb') as f:
+        f.write(TEST_FILE_2_CONTENT)
+    with open(tmpdir.join('test.txt').strpath, 'wb') as f:
+        f.write(bytes(TEST_FILE_3_CONTENT))
+    with open(tmpdir.join('empty-file').strpath, 'w'):
+        pass
+    return tmpdir
+
+
+LIST_ARTIFACTS_PROTO_RESPONSE = [FileInfo(path='test/a.txt', is_dir=False, file_size=100),
+                                 FileInfo(path='test/dir', is_dir=True, file_size=0)]
+
+LIST_ARTIFACTS_SINGLE_FILE_PROTO_RESPONSE = [FileInfo(path='a.txt', is_dir=False, file_size=0)]
+MOCK_AZURE_SIGNED_URI = "this_is_a_mock_sas_for_azure"
+MOCK_RUN_ID = 'MOCK-RUN'
+
+
+class TestDatabricksArtifactRepository(object):
+    def test_init_validation_and_cleaning(self):
+        repo = get_artifact_repository('dbfs:/databricks/mlflow-tracking/EXP/RUN/artifact')
+        assert repo.artifact_uri == 'dbfs:/databricks/mlflow-tracking/EXP/RUN/artifact'
+        with pytest.raises(MlflowException):
+            DatabricksArtifactRepository('s3://test')
+        with pytest.raises(MlflowException):
+            DatabricksArtifactRepository('dbfs:/databricks/mlflow/EXP/RUN/artifact')
+
+    @pytest.mark.parametrize("artifact_path,expected_location", [
+        (None, 'test.txt'),
+        ('output', 'output/test.txt'),
+        ('', 'test.txt'),
+    ])
+    def test_log_artifact(self, databricks_artifact_repo, test_file, artifact_path, expected_location):
+        with mock.patch(DATABRICKS_ARTIFACT_REPOSITORY + '._get_write_credentials') as write_credentials_mock, \
```

Comment:
Can we verify that the `_get_write_credentials` method is called with the expected run ID? This will indirectly test our run ID parsing.

Reply:
Its already present on line 80. Is there something I am missing?

Reply:
Sorry! Totally missed this. Looks great as is!